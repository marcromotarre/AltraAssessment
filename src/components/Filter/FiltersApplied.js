import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CandleSVG from '../../svg/candle.svg';
import WeightSVG from '../../svg/weight.svg';
import HeightSVG from '../../svg/height.svg';

import {
  isMinMaxFilterApplied,
  getFilterMinMaxString,
  isAgeOrWeightOrHeightFilterApplied,
} from './../../utils/filters';
import ColoredHair from '../User/ColoredHair';

const FiltersAppliedContainer = styled.section`
  padding: 10px 0px;
  background: #f7f8fd;
  display: grid;
  grid-template-columns: 100%;
`;

const PrimaryFilters = styled.div`
  display: flex;
  justify-content: space-evenly;
  align-items: center;
`;
const SecondaryFilters = styled.div`
  padding-top: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  svg {
    margin: 0px 10px;
  }
`;
const Filter = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  label {
    padding-left: 10px;
    font-family: 'HelveticaNeue-Thin';
    color: #000;
    font-size: 12px;
  }
`;

const AgeWeightHeightFilters = ({ age, weight, height }) => {
  return (
    <>
      {isMinMaxFilterApplied(age) && (
        <Filter>
          <CandleSVG width={14.63} />
          <label>{getFilterMinMaxString(age)}</label>
        </Filter>
      )}
      {isMinMaxFilterApplied(weight) && (
        <Filter>
          <WeightSVG width={17.7} />
          <label>{getFilterMinMaxString(weight)}</label>
        </Filter>
      )}
      {isMinMaxFilterApplied(height) && (
        <Filter>
          <HeightSVG height={17.7} />
          <label>{getFilterMinMaxString(height)}</label>
        </Filter>
      )}
    </>
  );
};

const HairFilter = ({ hair_colors }) => {
  return (
    <>
      {hair_colors.map((hair_color) => (
        <ColoredHair key={hair_color} color={hair_color} width={14.24} />
      ))}
    </>
  );
};

const FiltersApplied = ({ age, weight, height, hair_colors }) => {
  return (
    <FiltersAppliedContainer data-testid="FiltersApplied">
      <PrimaryFilters>
        {isAgeOrWeightOrHeightFilterApplied({
          age,
          height,
          weight,
        }) && (
          <AgeWeightHeightFilters age={age} weight={weight} height={height} />
        )}
        {!isAgeOrWeightOrHeightFilterApplied({ age, height, weight }) &&
          hair_colors.length > 0 && <HairFilter hair_colors={hair_colors} />}
      </PrimaryFilters>
      <SecondaryFilters>
        {isAgeOrWeightOrHeightFilterApplied({ age, height, weight }) &&
          hair_colors.length > 0 && <HairFilter hair_colors={hair_colors} />}
      </SecondaryFilters>
    </FiltersAppliedContainer>
  );
};

FiltersApplied.propTypes = {
  age: PropTypes.exact({
    min: PropTypes.string,
    max: PropTypes.string,
    range: PropTypes.bool,
  }),
  weight: PropTypes.exact({
    min: PropTypes.string,
    max: PropTypes.string,
    range: PropTypes.bool,
  }),
  height: PropTypes.exact({
    min: PropTypes.string,
    max: PropTypes.string,
    range: PropTypes.bool,
  }),
  hair_colors: PropTypes.arrayOf(PropTypes.string),
  onClickFilterIcon: PropTypes.func,
};

FiltersApplied.defaultProps = {
  age: { min: '', max: '', range: false },
  weight: { min: '', max: '', range: false },
  height: { min: '', max: '', range: false },
  hair_colors: [],
  onClickFilterIcon: () => {},
};

export default FiltersApplied;
