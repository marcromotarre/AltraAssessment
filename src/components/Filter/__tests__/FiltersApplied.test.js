import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import FiltersApplied from '../FiltersApplied';

/** this component should:
 *  - render FiltersApplied with:
 *      - filter icon button
 *          - onclick calls props function
 *      - filters
 *          - age (optional) (with test formating)
 *          - weight (optional) (with test formating)
 *          - height (optional) (with test formating)
 *          - hair color filter (optional)
 */

const data_range_empty = {
  min: '',
  max: '',
  range: true,
};

const data_empty = {
  min: '',
  max: '',
  range: true,
};

const data_range_min_max = {
  min: '34',
  max: '128',
  range: true,
};

const data_min_max = {
  min: '34',
  max: '128',
  range: false,
};

const data_max = {
  min: '',
  max: '128',
  range: false,
};

const data_min = {
  min: '34',
  max: '',
  range: false,
};

const data_range_min = {
  min: '34',
  max: '',
  range: true,
};

const data_range_max = {
  min: '',
  max: '128',
  range: true,
};

const colors_full = ['Red', 'Pink', 'Black'];

const props = {
  age: data_range_min_max,
  weight: data_range_min,
  height: data_range_max,
  hair_colors: colors_full,
  onClickFilterIcon: jest.fn(),
};

const emptyAgeHeightWeightProps = {
  age: data_range_empty,
  weight: data_empty,
  height: data_range_empty,
  hair_colors: colors_full,
  onClickFilterIcon: jest.fn(),
};

const emptyHairColorsProps = {
  age: data_range_min_max,
  weight: data_range_min_max,
  height: data_range_min_max,
  hair_colors: [],
  onClickFilterIcon: jest.fn(),
};

describe('FiltersApplied Component tests', () => {
  it('FiltersApplied Snapshot', () => {
    const tree = renderer.create(<FiltersApplied {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render age, weight, height filters', () => {
    const wrapper = shallow(<FiltersApplied {...props} />);
    const ageWeightHEightFilters = wrapper.childAt(0);
    expect(ageWeightHEightFilters.type().displayName).toBe('styled.div');
    expect(ageWeightHEightFilters.props().children[0].props.age).toBe(
      data_range_min_max
    );
    expect(ageWeightHEightFilters.props().children[0].props.weight).toBe(
      data_range_min
    );
    expect(ageWeightHEightFilters.props().children[0].props.height).toBe(
      data_range_max
    );
  });

  it('should render hair_filter', () => {
    const wrapper = shallow(<FiltersApplied {...props} />);
    const hairFilters = wrapper.childAt(1).childAt(0);
    expect(hairFilters.props().hair_colors).toBe(colors_full);
  });

  it('should not render AgeWeightHeightFilters if data_range_empty or data_empty in  age, height and weight. also painr HairFilter in PrimarySection', () => {
    const wrapper = shallow(<FiltersApplied {...emptyAgeHeightWeightProps} />);
    const hairFilters = wrapper.childAt(0).childAt(0);
    expect(hairFilters.props().hair_colors).toBe(colors_full);
  });

  it('should not render hairs in HairFilter if hair_colors is empty', () => {
    const wrapper = shallow(<FiltersApplied {...emptyHairColorsProps} />);
    const ageWeightHeightFilters = wrapper.childAt(0);
    expect(ageWeightHeightFilters.type().displayName).toBe('styled.div');
    expect(ageWeightHeightFilters.props().children[0].props.age).toBe(
      data_range_min_max
    );
    const hairFilters = wrapper.childAt(1);
    expect(hairFilters.type().displayName).toBe('styled.div');
    expect(hairFilters.children().length).toBe(0);
  });
});
