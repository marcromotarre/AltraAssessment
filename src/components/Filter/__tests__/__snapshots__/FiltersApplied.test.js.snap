// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersApplied Component tests FiltersApplied Snapshot 1`] = `
<section
  className="sc-AxirZ fAmXjD"
  data-testid="FiltersApplied"
>
  <div
    className="sc-AxiKw jwJFgJ"
  >
    <div
      className="sc-AxhUy ilSPof"
    >
      <svg
        width={14.63}
      />
      <label>
        34 - 128
      </label>
    </div>
    <div
      className="sc-AxhUy ilSPof"
    >
      <svg
        width={17.7}
      />
      <label>
        34
      </label>
    </div>
    <div
      className="sc-AxhUy ilSPof"
    >
      <svg
        height={17.7}
      />
      <label>
        128
      </label>
    </div>
  </div>
  <div
    className="sc-AxhCb jedUDk"
  >
    <svg
      className="sc-AxjAm dtmqRL"
      color="Red"
      width={14.24}
    />
    <svg
      className="sc-AxjAm fSpQWU"
      color="Pink"
      width={14.24}
    />
    <svg
      className="sc-AxjAm fvaMcT"
      color="Black"
      width={14.24}
    />
  </div>
</section>
`;
